packages/kit/src/runtime/server/respond.js

```diff
@@ -375,12 +378,12 @@ export async function respond(request, options, manifest, state) {
 					);
 				} else if (route.endpoint && (!route.page || is_endpoint_request(event))) {
 					response = await render_endpoint(event, await route.endpoint(), state);
-				} else if (route.page) {
+				} else if (route.page && page_methods.includes(method)) {
 					response = await render_page(event, route.page, options, manifest, state, resolve_opts);
 				} else {
-					// a route will always have a page or an endpoint, but TypeScript
-					// doesn't know that
-					throw new Error('This should never happen');
+					// not a valid endpoint request and no page exists.
+					const mod = route.endpoint ? await route.endpoint() : {};
+					return method_not_allowed(mod, method);
```

Comment:
Is this the correct outcome for `OPTIONS` requests? Should this be a 200 for non-cors page/endpoint requests with the allowed methods and 405 for all others? Reasoning:
- we don't want to get into "cors options request automatic handling" territory, so we leave that out (405) and people need to add their own logic if they want to handle that. That's ok for 99,9% of cases because
- pages are basically never called with cors (what's even a use case for this? when would this happen?)
- endpoints can add their own `OPTIONS` handler

Reply:
I'm not sure what the correct behaviour is so I'm open to implementing this.
That being said, I'm not sure how this will interact with the `OPTIONS` requests from Vercel https://github.com/sveltejs/kit/issues/9217#issue-1599806799. Is it okay to return a 405 to those requests?

Reply:
No idea honestly 😅 judging from the `OPTIONS` request headers it doesn't seem to be a CORS request so should be fine if we were to relax the "no allowed" case as I proposed. But I definitly should test it / we should think more about the implications of this.

Reply:
copying my message from discord for public visibility:

I've uploaded a prebuilt project to Vercel with the patch from https://github.com/sveltejs/kit/pull/9752 at https://kit-pr-9752-s3812497.vercel.app/

I think my code comment was misleading, and I had forgotten, OPTIONS requests are already returning 405 when requesting the page.  In turn, Vercel outputs a warning log for these 405 requests. There don't seem to be any noticeable negative effects for now (Vercel comments are still working).
~but it could be misleading when viewing project analytics regarding requests?~
(EDIT: Vercel doesn't measure request status code frequency)

Reply:
will implement returning a 200 instead of a 405 for page `OPTIONS` requests